components/hab/install.ps1

```diff
@@ -0,0 +1,224 @@
+<#
+.SYNOPSIS
+Installs the Habitat 'hab' program.
+
+Authors: The Habitat Maintainers <humans@habitat.sh>
+
+.DESCRIPTION
+This script builds habitat components and ensures that all necesary prerequisites are installed.
+
+.Parameter Channel
+Specifies a channel
+
+.Parameter Version
+Specifies a version (ex: 0.75.0, 0.75.0/20190219232208)
+#>
+
+param (
+    [Alias("c")]
+    [string]$Channel="stable",
+    [Alias("v")]
+    [string]$Version
+)
+
+$ErrorActionPreference="stop"
+
+Function Get-Version($version, $channel) {
+    $jsonFile = Join-Path (Get-WorkDir) "version.json"
+
+    $version = $version.Replace("/", "-")
+    if(!$version) {
+        $version = "%24latest"
+    } elseif([System.Reflection.Assembly]::LoadWithPartialName("System.Web.Extensions")) {
+        # Using .Net json serializer instead of Convert-FromJson (in PS v3 (win 2012) and greater)
+        # because the .Net strategy will work on any system with .Net 3.5 and up
+        Write-Host "Determining fully qualified version of package for '$version'"
+        Get-File "https://api.bintray.com/packages/habitat/$channel/hab-x86_64-windows" $jsonFile
+        $ser = New-Object System.Web.Script.Serialization.JavaScriptSerializer
+        $versions = $ser.DeserializeObject((Get-Content $jsonFile)).versions
+        $rev = $versions | ? { $_.StartsWith($version) }
+        if(!$rev) {
+            $e =  "Version '$version' could not used or version doesn't exist."
+            $e += " Please provide a simple version like: '0.15.0'"
+            $e += " or a fully qualified version like: '0.15.0/20161222203215'."
+            Write-Error $e
+        } else {
+            Write-Host "Using fully qualified Bintray version string of: $rev"
+            $version = $rev
+        }
+    } else {
+        # No ConvertFrom-Json support so we must be on unpatched PSv2
+        if($version -match "^\d\.\d\d\.\d-\d\d\d\d\d\d\d\d\d\d\d\d\d\d`$") {
```

Comment:
Is this matching a string like `0.75.0/20190219232208`?

If so, this is probably too restrictive... nothing guarantees that our major version will only ever be one digit, or our minor version two digits and our patch one digit (in fact, as we move to "release with each merge", it's going to be fairly common to see triple-digit minor and patch versions).

Does Powershell regex have syntax for quantifying numbers of repetitions? Something like `\d+\.\d+\.\d+/\d{12}` would be much more readable, for instance.

Also, is this intending to match only something like `0.75.0/20190219232208`, or should it also try to match `0.75.0`, too?

Reply:
ah good catch. your `\d+\.\d+\.\d+/\d{12}` works. Thanks!

Reply:
the intent is to make sure it is fully qualified. If we cant parse the json file to find the actual release we are trusting the user is giving it to us.

Reply:
That `12`s gonna break in 7,981 years

Reply:
The oft-overlooked Y10K problem...